AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1876,13 +1880,22 @@ - (ASLayout *)calculateLayoutThatFits:(ASSizeRange)constrainedSize
 
 - (CGSize)calculateSizeThatFits:(CGSize)constrainedSize
 {
+  __ASDisplayNodeCheckForLayoutMethodOverrides;
+    
   ASDN::MutexLocker l(_propertyLock);
   return _preferredFrameSize;
 }
 
 - (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
 {
+  __ASDisplayNodeCheckForLayoutMethodOverrides;
+
   ASDN::MutexLocker l(_propertyLock);
+  
+  if (_layoutSpecBlock != nil) {
+    return _layoutSpecBlock(constrainedSize);
```

Comment:
Very cool, extremely clean!  I think you should compare the block against NULL rather than nil.  It's probably fine to just do if (_layoutSpecBlock), but != NULL is fine too (there are some style benefits either way so it's your choice)